pkg/operator/controller/ingress/status.go

```diff
@@ -130,10 +180,16 @@ func computeIngressDegradedCondition(deployment *appsv1.Deployment) operatorv1.O
 // if the provided values should be considered equal for the purpose of determining
 // whether an update is necessary, false otherwise.
 func ingressStatusesEqual(a, b operatorv1.IngressControllerStatus) bool {
+	if a.ObservedGeneration != b.ObservedGeneration {
+		return false
+	}
 	if !conditionsEqual(a.Conditions, b.Conditions) || a.AvailableReplicas != b.AvailableReplicas ||
 		a.Selector != b.Selector {
 		return false
 	}
+	if !reflect.DeepEqual(a.TLSProfile, b.TLSProfile) {
```

Comment:
Isn't there a `semantic.DeepEqual` around that is supposed to be more appropriate for API types for Reasons?

Reply:
Grepping the Origin and Kubernetes code, `reflect.DeepEqual` appears to be widely used, `semantic.DeepEqual` not so much.  If I recall correctly, `semantic.DeepEqual` might be useful for some API objects for which certain fields need to be ignored or sorted, but I do not know a reason not to use `reflect.DeepEqual` here.  If anything, I would consider go-cmp (which we use elsewhere in this operator).